pkgs/servers/plex/default.nix

```diff
@@ -1,81 +1,52 @@
-{ stdenv, fetchurl, rpmextract, glibc
-, dataDir ? "/var/lib/plex" # Plex's data directory must be baked into the package due to symlinks.
-, enablePlexPass ? false
-}:
+{ stdenv, fetchurl, buildFHSUserEnv, runCommand, writeScript, rpmextract }:
 
 let
-  plexPass = throw "Plex pass has been removed at upstream's request; please unset nixpkgs.config.plex.pass";
-  plexpkg = if enablePlexPass then plexPass else {
-    version = "1.13.0.5023";
-    vsnHash = "31d3c0c65";
-    sha256 = "b94d571bcd47bc3201fd7f54b910f5693cabd1b543d5d0b4705668b7b7c24602";
-  };
-
-in stdenv.mkDerivation rec {
-  name = "plex-${version}";
-  version = plexpkg.version;
-  vsnHash = plexpkg.vsnHash;
-  sha256 = plexpkg.sha256;
+  name = "plexmediaserver-${version}";
+  version = "1.13.2.5154-fd05be322";
 
+  # https://downloads.plex.tv/plex-media-server/1.13.2.5154-fd05be322/plexmediaserver-1.13.2.5154-fd05be322.x86_64.rpm
   src = fetchurl {
-    url = "https://downloads.plex.tv/plex-media-server/${version}-${vsnHash}/plexmediaserver-${version}-${vsnHash}.x86_64.rpm";
-    inherit sha256;
+    url = "https://downloads.plex.tv/plex-media-server/${version}/${name}.x86_64.rpm";
+    sha256 = "09hy9xhhv17jbzplyls13xrzaxndlc278amp6k3w8q4j6wpsi6np";
   };
 
-  buildInputs = [ rpmextract glibc ];
+  shellScript = content: writeScript "script" "#!${stdenv.shell}\n${content}";
 
-  phases = [ "unpackPhase" "installPhase" "fixupPhase" "distPhase" ];
-
-  unpackPhase = ''
-    rpmextract $src
+  app = runCommand name { outputs = [ "out" "db" ]; } ''
+    ${rpmextract}/bin/rpmextract ${src} && mv usr $out
+    f=$out/lib/plexmediaserver/Resources/com.plexapp.plugins.library.db
+    cat $f > $db
+    ln -fs /db $f
   '';
+in
 
-  installPhase = ''
-    install -d $out/usr/lib
-    cp -dr --no-preserve='ownership' usr/lib/plexmediaserver $out/usr/lib/
-
-    # Now we need to patch up the executables and libraries to work on Nix.
-    # Side note: PLEASE don't put spaces in your binary names. This is stupid.
-    for bin in "Plex Media Server"              \
-               "Plex DLNA Server"               \
-               "Plex Media Scanner"             \
-               "Plex Media Server Tests"        \
-               "Plex Relay"                     \
-               "Plex Script Host"               \
-               "Plex Transcoder"                \
-               "Plex Tuner Service"             ; do
-      patchelf --set-interpreter "${glibc.out}/lib/ld-linux-x86-64.so.2" "$out/usr/lib/plexmediaserver/$bin"
-      patchelf --set-rpath "$out/usr/lib/plexmediaserver" "$out/usr/lib/plexmediaserver/$bin"
-    done
+buildFHSUserEnv {
```

Comment:
Why use buildFHSUserEnv instead of patching a few binaries, doesn't this make accessing media way more complicated then it should be?

Reply:
Because you can't patch binaries that Plex downloads at runtime.